scripts/zephyr_module.py

```diff
@@ -374,6 +374,7 @@ def kconfig_snippet(meta, path, kconfig_file=None, blobs=False, taint_blobs=Fals
     name_sanitized = meta['name-sanitized']
 
     snippet = [f'menu "{name} ({path.as_posix()})"']
+    snippet += [f'ZEPHYR_{name_sanitized.upper()}_MODULE_DIR := {path.as_posix()}']
```

Comment:
the snippet created by `kconfig_snippet()` is creating a Kconfig file similar to the `Kconfig.modules` file created here:
https://github.com/zephyrproject-rtos/zephyr/blob/e98920d6958525e74f14b0250b56241388c88bf4/cmake/modules/zephyr_module.cmake#L45
https://github.com/zephyrproject-rtos/zephyr/blob/e98920d6958525e74f14b0250b56241388c88bf4/cmake/modules/zephyr_module.cmake#L60

whereas the `ZEPHYR_<name>_MODULE_DIR` is exported to Kconfig as env var here:
https://github.com/zephyrproject-rtos/zephyr/blob/e98920d6958525e74f14b0250b56241388c88bf4/cmake/modules/kconfig.cmake#L109

Perhaps if doing this in the generated Kconfig file here, then we should update `cmake/modules/kconfig.cmake` also, so that at least compliance and kconfig.cmake are somewhat aligned in how things are done.